src/components/Quote.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from 'react';
+import './Quote.css';
+
+const uniqueID = '8YHMipVLjkYiF4KOWb7GUYqkO6VXXgcUK3qQJEAI';
+
+function Quote() {
+  const apiUrl = 'https://api.api-ninjas.com/v1/quotes';
+  const [quoteText, setQuoteText] = useState('');
+  const [authorText, setAuthor] = useState('');
+  const [categoryText, setCategory] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    const fetchQuote = async () => {
+      try {
+        const response = await fetch(apiUrl, {
+          headers: {
+            'X-Api-Key': uniqueID,
+          },
+        });
+        if (!response.ok) {
+          throw new Error(
+            `API request failed with status code ${response.status}.`,
+          );
+        }
+        const data = await response.json();
+        setQuoteText(data[0].quote);
+        setAuthor(data[0].author);
+        setCategory(data[0].category);
+      } catch (error) {
+        setHasError(true);
+      }
+      setIsLoading(false);
+    };
+
+    fetchQuote();
+  }, []); // Empty dependency array to run only once on initial component mount
+
+  const fetchQuote = async () => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(apiUrl, {
+        headers: {
+          'X-Api-Key': uniqueID,
+        },
+      });
+      if (!response.ok) {
+        throw new Error(
+          `API request failed with status code ${response.status}.`,
+        );
+      }
+      const data = await response.json();
+      setQuoteText(data[0].quote);
+      setAuthor(data[0].author);
+      setCategory(data[0].category);
+    } catch (error) {
+      setHasError(true);
+    }
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    const timer = setInterval(fetchQuote, 5000); // Change quote every 5 seconds (adjust as needed)
+
+    return () => clearInterval(timer); // Clear the timer on component unmount
+  }, []); // Empty dependency array to run only once on initial component mount
```

Comment:
- [ ] Great job working on this milestone! I really liked the refreshing timer. However, remember to keep your code clean, and follow the DRY principle. If you look closely, you are repeating the same `fetchQuote` function inside the `useEffect` and outside, the only difference is the `setIsLoading(true)` at the top. One thing to notice here is that both effects have the same dependency array, which means they run under the same conditions, on component mount.

Taking this is mind, you could easily merge both effects, ending up with something like this:

```suggestion
  useEffect(() => {
    const fetchQuote = async () => {
      setIsLoading(true);
      try {
        const response = await fetch(apiUrl, {
          headers: {
            'X-Api-Key': uniqueID,
          },
        });
        if (!response.ok) {
          throw new Error(
            `API request failed with status code ${response.status}.`,
          );
        }
        const data = await response.json();
        setQuoteText(data[0].quote);
        setAuthor(data[0].author);
        setCategory(data[0].category);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };

    fetchQuote();
    const timer = setInterval(fetchQuote, 5000); // Change quote every 5 seconds (adjust as needed)
    return () => clearInterval(timer); // Clear the timer on component unmount
  }, []); // Empty dependency array to run only once on initial component mount
```

Please consider applying this change! 🤓 🔧

Reply:
You are really observant. I admire your attention to detail. Thanks for that critical observation!